src/main/java/dev/callumherr/modding/ducky_lib/block/entity/AbstractMultiBlock.java

```diff
@@ -0,0 +1,213 @@
+                case LEFT_RIGHT: // mirror across the x-axis (flip Z)
+                    return state.setValue(Z, 2 - state.getValue(Z));
+                case FRONT_BACK: // mirror across the z-axis (flip X)
+                    return state.setValue(X, 2 - state.getValue(X));
+                default:
+                    return state;
+            }
+        } else {
+            return super.mirror(state, mirror);
+        }
+    }
+
+    /**
+     * Creates a new block entity at the given position and state.
+     *
+     * <p>This method is abstract and must be implemented by subclasses to return an appropriate block entity
+     * for the multi-block structure.</p>
+     *
+     * @param pos   The position of the block.
+     * @param state The block state at the position.
+     * @return The new block entity instance.
+     */
+    @Nullable
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pos, BlockState state) {
+        return createBlockEntity(pos, state);
+    }
+
+    /**
+     * Abstract method to create a block entity for the multi-block structure.
+     *
+     * @param pos   The position of the block.
+     * @param state The block state at the position.
+     * @return The block entity instance.
+     */
+    public abstract BlockEntity createBlockEntity(BlockPos pos, BlockState state);
+
+    /**
+     * Returns a ticker for the block entity at the given position, if applicable.
+     *
+     * <p>This method is abstract and must be implemented by subclasses to return an appropriate ticker
+     * for the multi-block structure's block entity.</p>
+     *
+     * @param level The level in which the block is present.
+     * @param state The block state.
+     * @param type  The block entity type.
+     * @param <T>   The type of block entity.
+     * @return The block entity ticker or null if no ticker is required.
+     */
+    @Nullable
+    @Override
+    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, BlockState state, BlockEntityType<T> type) {
+        return createTicker(level, state, type);
+    }
+
+    /**
+     * Abstract method to create a ticker for the block entity in the multi-block structure.
+     *
+     * @param level The level in which the block is present.
+     * @param state The block state.
+     * @param type  The block entity type.
+     * @param <T>   The type of block entity.
+     * @return The block entity ticker or null if no ticker is required.
+     */
+    protected abstract <T extends BlockEntity> BlockEntityTicker<T> createTicker(Level level, BlockState state, BlockEntityType<T> type);
+
+    /**
+     * Sends a message to nearby players when the multi-block structure is formed.
+     * Not used since debugging is complete. In the future, it may be used.
+     * @param pos   The position of the block.
+     * @param state The block state.
+     */
+    private void sendMultiblockFormedMessage(BlockPos pos, BlockState state) {
```

Comment:
Can be turned into an onMultiBlockFormed method

Reply:
hmm ye